src/biome/text/dataset.py

```diff
@@ -341,6 +347,33 @@ def instance_unpickler(instances_col_name: str) -> Iterable[Instance]:
 
         return instance_unpickler
 
+    def _create_fingerprint_for_instance_dataset(self, pipeline: "Pipeline") -> str:
+        """Creates a fingerprint for the instance dataset to take advantage of datasets caching mechanism
+
+        The fingerprint is based on:
+        - the fingerprint of the previous dataset
+        - the tokenizer config
+        - the indexer config of the features
+        - the biome__version__, allennlp__version__ and spaCy__version__ just to be completely sure!
+        """
+        hasher = Hasher()
+        # add the previous fingerprint to the hash
+        hasher.update(self.dataset._fingerprint)  # necessary evil ...
```

Comment:
The underlying fingerprint isn't built at map operation level, but at whole dataset level?

Reply:
Yes, so basically every dataset instance has a fingerprint. when applying a map operation, a new fingerprint is calculated based on the dataset's fingerprint and the arguments of the map operation. If a dataset with this new fingerprint is present in the cache, this dataset will be returned and the map operation is omitted.